pkg/volume/csi/csi_plugin.go

```diff
@@ -59,9 +62,48 @@ func ProbeVolumePlugins() []volume.VolumePlugin {
 // volume.VolumePlugin methods
 var _ volume.VolumePlugin = &csiPlugin{}
 
+type csiDriver struct {
+	driverName     string
+	driverEndpoint string
+}
+
+// csiDrivers map keep track of all registered CSI drivers on the node and their
+// corresponding sockets
+var csiDrivers sync.Map
+
+var lm labelmanager.LabelManagerInterface
+
+// RegistrationCallback is called by kubelet's plugin watcher upon detection
+// of a new registration socket opened by CSI Driver registrar side car.
+func RegistrationCallback(pluginName string, endpoint string, versions []string, socketPath string) (error, chan bool) {
+
+	glog.Infof("Callback from kubelet with plugin name: %s endpoint: %s versions: %s socket path: %s",
+		pluginName, endpoint, strings.Join(versions, ","), socketPath)
+
+	// Storing endpoint of newly registered CSI driver into the map, where CSI driver name will be the key
+	// all other CSI components will be able to get the actual socket of CSI drivers by its name.
+	if endpoint == "" {
```

Comment:
Should this be a check inside of the plugin watcher module itself?
Actually shouldn't this be a fail?

Reply:
Well my question was more general, this behavior seems like it should:
- either be forbidden (i.e: All plugins should specify their endpoint)
- or it should be handled by the pluginwatcher API so that all consumers of the pluginwatcher API have the same behavior.

In both case my opinion is that you shouldn't have to check that the interactions with the pluginwatcher api should provide you strong guarantees that the parameters are somewhat correct.

In particular I'm thinking that the goal should be to have a consistent registration mechanism for all plugins APIs (CSI, device plugin, ...). Thus my reasoning about this being handled by the pluginwatcher API.

What do you think?

Reply:
it is 48 hours before the code freeze, if you feel that something is not addressed, mind you that the feature is working as expected now, please file the issue and I will be happy to discuss it in the followup PR after the code freeze.

Reply:
Will do, I don't want this to be blocking for the code freeze

Reply:
Plugin Watcher should be doing this. Let's open a bug to move this logic to plugin watcher.